Some Monday patches

thinkyhead opened the pull request:
- Fix `ubl::g29_what_command` - removing `ubl.` prefix from member references
- Clean up trailing whitespace and tabs left in source (please change editor settings!)
- Finished adding `Q` parameter to `M421` (bilinear and MBL)

thinkyhead committed:
Fix up M421 and some comments

thinkyhead committed:
Patch lcd_implementation_status_screen compiler warning

thinkyhead committed:
Remove "ubl." prefix from calls within ubl method

thinkyhead committed:
UBL tabs, whitespace, spelling, etc.

Roxy-3D commented:
> Remove parsing of Q parameter errantly added to M421 (bilinear)

Shouldn't the Bi-Linear mesh be able to have its mesh points shifted by an amount and not just set to an absolute number?

thinkyhead commented:
The code to parse the parameter is there, but the code to apply the shift was never added. If that was the aim, it should have been applied to all the `M421` implementations, including MBL as well.

thinkyhead commented:
Ok… I've added '`Q`' to all three.

Roxy-3D commented:
The original logic for 'Q' did not do the original Mesh Leveling system...  But it handled 'Q' correctly:
```cpp
  /**
   * M421: Set a single Mesh Bed Leveling Z coordinate
   *
   *   M421 I<xindex> J<yindex> Z<linear>
   *   or
   *   M421 I<xindex> J<yindex> Q<offset>
   */
  inline void gcode_M421() {
    int8_t px = 0, py = 0;
    float z = 0;
    bool hasI, hasJ, hasZ, hasQ;
    if ((hasI = code_seen('I'))) px = code_value_linear_units();
    if ((hasJ = code_seen('J'))) py = code_value_linear_units();
    if ((hasZ = code_seen('Z'))) z = code_value_linear_units();
    if ((hasQ = code_seen('Q'))) z = code_value_linear_units();

    if (!hasI || !hasJ || (hasQ && hasZ) || (!hasQ && !hasZ)) {
      SERIAL_ERROR_START;
      SERIAL_ERRORLNPGM(MSG_ERR_M421_PARAMETERS);
      return;
    }

    if (WITHIN(px, 0, GRID_MAX_POINTS_X - 1) && WITHIN(py, 0, GRID_MAX_POINTS_Y - 1)) {
      if (hasZ) { // doing an absolute mesh value
        #if ENABLED(AUTO_BED_LEVELING_UBL)
          ubl.z_values[px][py] = z;
        #else
          z_values[px][py] = z;
          #if ENABLED(ABL_BILINEAR_SUBDIVISION)
            bed_level_virt_interpolate();
          #endif
        #endif
      } 
      else { // doing an offset of a mesh value
        #if ENABLED(AUTO_BED_LEVELING_UBL)
          ubl.z_values[px][py] += z;
        #else
          z_values[px][py] += z;
          #if ENABLED(ABL_BILINEAR_SUBDIVISION)
            bed_level_virt_interpolate();
          #endif
        #endif
      }
    }
    else { // bad indexes were specified for the mesh point
      SERIAL_ERROR_START;
      SERIAL_ERRORLNPGM(MSG_ERR_MESH_XY);
    }
  }
#endif
```

thinkyhead commented:
It's twice as fabulous now as it was then.

thinkyhead committed:
Reduce find_closest_mesh_point_of_type a little

thinkyhead committed:
UBL G29: replace 'O' with 'T'

Since we already used 'T' for `G29` Topology Report in the past, and since 'T' is available…

thinkyhead commented:
@Roxy-3D I'd like to propose using the letter '`T`' for `G29` **T**opology Map (instead of `O`) since that's what we have been using up to now, and serendipitously '`T`' is available. The last commit amends that code letter, if that is copacetic.

Roxy-3D commented:
I'm OK with T for the Topograhpical Map.   But let's not cause Tannoo to jump through hoops again.   A couple quick edits in ultraLCD.cpp will get his stuff fixed up.

Mean while...   I have a more serious issue with #6705 because I fixed some problems and they conflicted with this (important) Pull Request that was not yet merged.

thinkyhead commented:
Not a hard conflict to resolve. You just changed some lines that it also changed. It was simple to resolve using the "Resolve" button. The PR will be merged soon.

Roxy-3D commented:
Thank you....

thinkyhead commented:
@Roxy-3D Forgot to mention…

This is pretty good: http://marlinfw.org/docs/gcode/G026.html
This needs some filling in: http://marlinfw.org/docs/gcode/G029-ubl.html
Of course, feel free to keep adding to: http://marlinfw.org/docs/features/unified_bed_leveling.html
